proposals/stage_variable_expressions/README.md

```diff
@@ -106,24 +126,146 @@ error messages. If we didn't explicitly identify expressions somehow, the only
+include:
+
+```
+if(<bool>, <if-value>, <else-value>)
+if(<bool>, <if-value>)
+
+not(<expression>)
+or(<expression>, <expression>, ...)
+and(<expression>, <expression>, ...)
+defined(<stage var>)
+
+str_starts_with(<expression>, <expression>)
+str_ends_with(<expression>, <expression>)
+
+```
+
+We do not propose allowing plugins to define additional functions for use
+in expressions.
+
+#### Grammar
+
+The following is a pseudo-BNF grammar for the proposed syntax. Within the BNF:
+
+- "{" and "}" are groupings of non-optional elements
+- "[" and "]" are groupings of optional elements.
+- "*" indicates zero-or-more of the associated elements.
+- "+" indicates one-or-more of the associated elements.
+
+```
+; A stage variable expression is an expression surrounded by backticks.
+<stage-variable-expression> = "`" <expression> "`"
+
+; An expression can be a stage variable reference, a literal, a function,
+; or a list of these elements.
+<expression_no_list> = <stage-variable>
+                     | <literal>
+                     | <function>
+<expression> = <expression_no_list>
+             | <list>
+
+; A stage variable reference is the name of the stage variable surrounded
+; by "${" and "}". Note that stage variable names must be a valid identifier.
+<identifier> = {<letter> | "_"} [<letter> | <digit> | "_"]*
+<stage-variable> = "${" <identifier> "}"
+
+; A literal can be one of the supported value types.
+<literal> = <boolean>
+          | <integer>
+          | <string>
+
+; Integer values
+<integer> = ["-"] <digit>+
+
+; Boolean values in expressions may take any of these forms. These were
+; chosen since they are the representations used in the two languages
+; that USD primarily supports: C++ and Python.
+<boolean> = "true" | "True" | "false" | "False"
+
+; String values may be single (') or double (") quoted.
+;
+; Note that strings may have stage variable references embedded in them.
+; During evaluation, these references will be substituted with the value
+; of the named stage variable. 
+;
+; Strings use backslashes (\) to escape delimiters, including the leading
+; "$" in a stage variable reference.
+<string> = <single-quoted-string>
+         | <double-quoted-string>
+
+<single-quoted-string> = "'" <single-quoted-string-contents> "'"
+<single-quoted-string-contents> = [<single-quoted-escaped-char> | <char> | <stage-variable>]*
+<single-quoted-escape-char> = "\" ["'" | "$" | "\"]
+
+<double-quoted-string> = '"' <double-quoted-string-contents> '"' 
+<double-quoted-string-contents> = [<double-quoted-escaped-char> | <char> | <stage-variable>]*
+<double-quoted-escape-char> = "\" ['"' | "$" | "\"]
+
+; A function is an identifier followed by a comma-separated list of
+; expressions, surrounded by "(" and ")".
+<function> = <identifier> "(" [<expression> ["," <expression>]*] ")"
+
+; A list is a comma-separated list of non-list expressions, surrounded by
+; "[" and "]".
+<list> = "[" [<expression_no_list> ["," <expression_no_list>]*] "]"
```

Comment:
In the enclosing "##Expressions" section, we state that we support "One-dimensional, homogeneous arrays of the above types."

But this `<list>` element looks like a heterogeneous container, as defined, and there is no mention of arrays.  Can you clarify, please?